src/main/java/ru/ayaz/UserMessageHandler.java

```diff
@@ -0,0 +1,142 @@
+package ru.ayaz;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+public class UserMessageHandler implements Runnable {
+    private Map<String, User> usersMap;
+    private Map<String, PrintWriter> outputWriterMap;
+    private Map<String, BufferedReader> inputReaderMap;
+    private BlockingQueue<UserMessage> messageQueue;
+    private MessageParser parser;
+
+    public UserMessageHandler(int queueSize) {
+        usersMap = new HashMap<String, User>();
+        outputWriterMap = new HashMap<String, PrintWriter>();
+        inputReaderMap = new HashMap<String, BufferedReader>();
+        messageQueue = new ArrayBlockingQueue(queueSize, true);
+        parser = new MessageParser();
+    }
+
+    public void enqueueMessage(UserMessage message) {
+        try {
+            messageQueue.put(message);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public boolean queueContainsMessage(UserMessage message) {
+        return messageQueue.contains(message);
+    }
+
+    public UserMessage takeMessage() throws InterruptedException {
+        return messageQueue.take();
+    }
+
+    public void addUserToMap(User user) {
+        usersMap.put(user.getNickname(), user);
+    }
+
+    public void addWriterToMap(String nickname, PrintWriter writer) {
+        outputWriterMap.put(nickname, writer);
+    }
+
+    public void addReaderToMap(String nickname, BufferedReader reader) {
```

Comment:
&#x1F34E; **Issue**
This method could be named better. Just from looking at the name question pops up: "What is this Map ?"